source/NVDAObjects/window/winword.py

```diff
@@ -1173,7 +1173,13 @@ def populateHeaderCellTrackerFromBookmarks(self,headerCellTracker,bookmarks):
 	_curHeaderCellTracker=None
 	def getHeaderCellTrackerForTable(self,table):
 		tableRange=table.range
-		if not self._curHeaderCellTrackerTable or not tableRange.isEqual(self._curHeaderCellTrackerTable.range):
+		# Sometimes there is a valid reference in _curHeaderCellTrackerTable,
+		# but we get a COMError when accessing the range (#6827)
+		try:
+			tableRangesEqual=tableRange.isEqual(self._curHeaderCellTrackerTable.range)
+		except (COMError, AttributeError):
+			tableRangesEqual=False
+		if not self._curHeaderCellTrackerTable or not tableRangesEqual:

```

Comment:
I belief the if not self._curHeaderCellTrackerTable will alreade be dealt with if the lines above raise an AttributeError. Honestly, I'd check for if not self._curHeaderCellTrackerTable before executing the try statement, which looks somewhat cleaner. I leave it up to you.

Reply:
You're right, the first part of this or statement is unneeded. I removed it and left the try/except intact. Removing the except AttributeError and solving this in an if statement would introduce another variable to hold state, or a level of nested if statements. I think having all the logic in a single try/except makes the code easier to read.